api/src/main/java/com/nicknathanjustin/streamercontracts/contracts/ContractModel.java

```diff
@@ -40,25 +40,63 @@
 
     private String description;
 
+    @Setter(AccessLevel.NONE)
     private Timestamp proposedAt;
 
+    @Setter(AccessLevel.NONE)
     private Timestamp acceptedAt;
 
+    @Setter(AccessLevel.NONE)
     private Timestamp declinedAt;
 
+    @Setter(AccessLevel.NONE)
     private Timestamp settlesAt;
 
+    @Setter(AccessLevel.NONE)
     private Timestamp expiredAt;
 
+    @Setter(AccessLevel.NONE)
     private Timestamp completedAt;
 
+    @Setter(AccessLevel.NONE)
     private Timestamp failedAt;
 
+    @Setter(AccessLevel.NONE)
     private Timestamp disputedAt;
 
     private boolean isCommunityContract;
 
-    private String state;
-
-    private String devnote;
+    @Setter(AccessLevel.NONE)
+    private ContractState state;
+
+    private String devNote;
+
+    public void setContractState(final ContractState newContractState) {
```

Comment:
Having business logic in a model class is an antipattern. I think we'd be best off moving this logic back to the service layer.

Reply:
Discussed offline. Leaving as is